include/zephyr/drivers/gnss.h

```diff
@@ -0,0 +1,497 @@
+/*
+ * Copyright (c) 2023 Trackunit Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file gnss.h
+ * @brief Public GNSS API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GNSS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GNSS_H_
+
+/**
+ * @brief GNSS Interface
+ * @defgroup gnss_interface GNSS Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+#include <zephyr/data/navigation.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** GNSS PPS modes */
+enum gnss_pps_mode {
+	/** PPS output disabled */
+	GNSS_PPS_MODE_DISABLED = 0,
+	/** PPS output always enabled */
+	GNSS_PPS_MODE_ENABLED = 1,
+	/** PPS output enabled from first lock */
+	GNSS_PPS_MODE_ENABLED_AFTER_LOCK = 2,
+	/** PPS output enabled while locked */
+	GNSS_PPS_MODE_ENABLED_WHILE_LOCKED = 3
+};
+
+/** API for setting fix rate */
+typedef int (*gnss_set_fix_rate_t)(const struct device *dev, uint32_t fix_interval_ms);
+
+/** API for getting fix rate */
+typedef int (*gnss_get_fix_rate_t)(const struct device *dev, uint32_t *fix_interval_ms);
+
+/**
+ * @brief GNSS periodic tracking configuration
+ *
+ * @note Setting either active_time or inactive_time to 0 will disable periodic
+ * function.
+ */
+struct gnss_periodic_config {
+	/** The time the GNSS will spend in the active state in ms */
+	uint32_t active_time_ms;
+	/** The time the GNSS will spend in the inactive state in ms */
+	uint32_t inactive_time_ms;
+};
+
+/** API for setting periodic tracking configuration */
+typedef int (*gnss_set_periodic_config_t)(const struct device *dev,
+					  const struct gnss_periodic_config *periodic_config);
+
+/** API for setting periodic tracking configuration */
+typedef int (*gnss_get_periodic_config_t)(const struct device *dev,
+					  struct gnss_periodic_config *periodic_config);
+
+/** GNSS navigation modes */
+enum gnss_navigation_mode {
+	/** Dynamics have no impact on tracking */
+	GNSS_NAVIGATION_MODE_ZERO_DYNAMICS = 0,
+	/** Low dynamics have higher impact on tracking */
+	GNSS_NAVIGATION_MODE_LOW_DYNAMICS = 1,
+	/** Low and high dynamics have equal impact on tracking */
+	GNSS_NAVIGATION_MODE_BALANCED_DYNAMICS = 2,
+	/** High dynamics have higher impact on tracking */
+	GNSS_NAVIGATION_MODE_HIGH_DYNAMICS = 3
+};
+
+/** API for setting navigation mode */
+typedef int (*gnss_set_navigation_mode_t)(const struct device *dev,
+					  enum gnss_navigation_mode mode);
+
+/** API for getting navigation mode */
+typedef int (*gnss_get_navigation_mode_t)(const struct device *dev,
+					  enum gnss_navigation_mode *mode);
+
+/**
+ * @name GNSS Systems
+ *
+ * @brief Systems contained in gnss_systems_t
+ *
+ * @{
+ */
```

Comment:
this is weird? but if you do want to have a nice "first-class" entry for GNSS Systems, you'd still need to add a @brief to the `gnss_system` enum - right now it's not documented, as the `@brief` applies to the name group.

Reply:
Changed it to
```
/** Systems contained in gnss_systems_t */
enum gnss_system {
```